pymc3/variational/updates.py

```diff
@@ -451,6 +491,20 @@ def adagrad(loss_or_grads=None, params=None, learning_rate=1.0, epsilon=1e-6):
 
     .. [2] Chris Dyer:
            Notes on AdaGrad. http://www.ark.cs.cmu.edu/cdyer/adagrad.pdf
+    
+    Examples
+    --------
+    >>> a = theano.shared(1.)
+    >>> b = a*2
+    >>> updates = adagrad(b, [a], learning_rate=.01)
+    >>> isinstance(updates, dict)
```

Comment:
Not sure the types are that useful to a user. I think how to use it practically or the actual outputs produced would be more valuable. Can't we set up a small optim problem?